src/inputs/SelectField.test.tsx

```diff
@@ -403,20 +403,28 @@ describe("SelectFieldTest", () => {
   it("supports boolean values properly", async () => {
     // Given a select field with boolean and an undefined values
     const onSelect = jest.fn();
-    const r = await render(
-      <SelectField
-        label="label"
-        value={true}
```

Comment:
Before this refactoring, this `value` could stay hard-coded and `ComboBoxBase` had its own internal version of values that could drift from the `value` prop.

But now, kinda on accident but I think it makes sense, the internal value is always derived from the `value` prop, so it behaves more like a controlled component.

Which is fine, afaict not a breaking change, but meant these tests that were hard-coding `value=true` needed to change.